great_expectations/expectations/metrics/query_metrics/query_data_source_table/query_data_source_table.py

```
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+from typing import Any, ClassVar, Dict
+
+from great_expectations.execution_engine import (
+    SqlAlchemyExecutionEngine,
+)
+from great_expectations.expectations.metrics.metric_provider import metric_value
+from great_expectations.expectations.metrics.query_metric_provider import (
+    InvalidParameterTypeError,
+    MissingParameterError,
+    QueryMetricProvider,
+)
+
+
+class QueryDataSourceTable(QueryMetricProvider):
+    metric_name = "query.data_source_table"
+    value_keys = ("query", "data_source_name")
+    data_source_name_param_name: ClassVar[str] = "data_source_name"
+
+    @classmethod
+    def _get_data_source_name_from_metric_value_kwargs(cls, metric_value_kwargs: dict) -> str:
+        data_source_name_param = cls.data_source_name_param_name
+        data_source_name: str | None = metric_value_kwargs.get(
```

Will we actually have `None` in these dictionaries if it's missing, or will we hit a KeyError before we can check for None and raise a `MissingParameterError`?